_security-plugin/configuration/ldap.md

```diff
@@ -641,3 +640,33 @@ authz:
         rolename: cn
         resolve_nested_roles: true
 ```
+
+### Connection pooling settings
+
+The directory server can maintain a pool of connections at the ready, assigning them when needed and returning them to the pool after a connection is closed. This arrangement can lower demands on the resources used to create connections, improve OpenSearch performance, and reduce load on the server. You can use the settings below to control the way the pooling is carried out. 
+
+Name | Description
+:--- | :---
+`pool.enabled` | Enables connection pooling. Set to `true` to enable.
+`pool.min_size` | Size of the pool at initialization. Also used as a lower limit when pruning.
+`pool.max_size` | Maximum size the pool can reach.
+`pool.pruning_period` | The interval in minutes at which the pruning implementation is executed. For example: when 5, the implementation is executed every five minutes. By default, the period is 5.
+`pool.idle_time` | The length of time elapsed, in minutes, after a connnection is considered idle. Once elapsed, the connection becomes a candidate for pruning from the pool. By default, idle time is 10.
```

Comment:
These settings are correct for v2.4+ otherwise they have different values `pruning.period` and `pruning.idleTime` in the older versions.

In 2.4+ legacy settings are still read and used and they are marked as deprecated

Reply:
@peternied I'm not sure I understand what the differences will be between versions -2.4 and versions 2.4+. Will all five connection pooling settings be available to versions before 2.4? If yes, I think you're saying the values will be different for the `pruning.period` and `pruning.idleTime` settings. If that's correct, what will they be?

Ultimately, I'm wondering what details in the documentation added here will lead users astray when they add these settings to config.yml and they're running versions before 2.4? If there are differences, it sounds like I'll need to create a separate PR for earlier versions.

Reply:
### Versions Less than 2.4.0 (e.g. 1.3, 2.1, 2.2, 2.3)
`pruning.period` and `pruning.idleTime` work, but were never documented except for the code

### Versions Equal or greater than 2.4.0
`pool.idle_time` and `pool.pruning_period` work.
`pruning.period` and `pruning.idleTime` work and spit out a deprecation warning in the OpenSearch log

I think you could document all the settings for the version to release in 2.4.0.  Then make another pull request that removes them targeting the older versions.  Someone familiar with this repo might have cleaner instructions or a workflow to follow